### PR TITLE
test(csharp): add e2e test suite

### PR DESCRIPTION
**Pull request by morganleroi**
## 🧭 What and Why

🎟 JIRA Ticket: DI-1766

### Changes included:
- Add e2e tests

### CODE REVIEW

**Review by Fluf22 — APPROVED** [review]
Nice work!

File: templates/csharp/tests/requests/requests.mustache [hunk 1]
```diff
@@ -42,7 +60,7 @@ public class {{client}}RequestTests
      Assert.Equal("{{{path}}}",req.Path);
      Assert.Equal("{{{method}}}",req.Method.ToString());
     {{#body}}
-        JsonAssert.EqualOverrideDefault("{{#lambda.escapeQuotes}}{{{body}}}{{/lambda.escapeQuotes}}", req.Body, new JsonDiffConfig(true));
+        JsonAssert.EqualOverrideDefault("{{#lambda.escapeQuotes}}{{{body}}}{{/lambda.escapeQuotes}}", req.Body, new JsonDiffConfig(false));
```
Comment (millotp) [inline templates/csharp/tests/requests/requests.mustache:63]: what does `JsonDiffConfig` do ? I can't find the doc

**Review by millotp — APPROVED** [review]
nice !

File: templates/csharp/tests/requests/requests.mustache (hunk 1 above)
Comment (morganleroi) [reply templates/csharp/tests/requests/requests.mustache:63]: The `JsonDiffConfig` allows additional properties to be added without error during assertion. (if `true`).

For the request with Echo mock, it was a mistake to allow additional properties to be added without failing. For the e2e, it's still necessary as some properties are added to the payload but we don't want to assert then (Request duration, .Processing time, ...). Does it make sense ?

**Pull request merged**